packages/zoltan2/test/color/ReadMatrixFromFile.hpp

```diff
@@ -0,0 +1,319 @@
+#ifndef __READMATRIXFROMFILE_HPP
+#define __READMATRIXFROMFILE_HPP
+#include<vector>
+template <typename global_ordinal_type, typename local_ordinal_type, typename scalar_type, typename map_type>
+void
+distribute (Teuchos::ArrayRCP<size_t>& myNumEntriesPerRow,
```

Comment:
We should think about having a common interface for binary and ASCII sparse matrix I/O, to avoid code duplication with Tpetra.